File: AdvantageFramework/Database/Core/Job.vb
Namespace Database.Core

    <Serializable()>
    Public Class Job
        Inherits AdvantageFramework.BaseClasses.BaseCore

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            Number
            OfficeCode
            ClientCode
            DivisionCode
            ProductCode
            Description
            IsActive
        End Enum

#End Region

#Region " Variables "

        Private _Number As Integer = Nothing
        Private _OfficeCode As String = Nothing
        Private _ClientCode As String = Nothing
        Private _DivisionCode As String = Nothing
        Private _ProductCode As String = Nothing
        Private _Description As String = Nothing
        Private _IsOpen As Nullable(Of Integer) = Nothing

#End Region

#Region " Properties "

        Public Property Number() As Integer
            Get
                Number = _Number
            End Get
            Set(ByVal value As Integer)
                _Number = value
            End Set
        End Property
        Public Property OfficeCode() As String
            Get
                OfficeCode = _OfficeCode
            End Get
            Set(ByVal value As String)
                _OfficeCode = value
            End Set
        End Property
        Public Property ClientCode() As String
            Get
                ClientCode = _ClientCode
            End Get
            Set(ByVal value As String)
                _ClientCode = value
            End Set
        End Property
        Public Property DivisionCode() As String
            Get
                DivisionCode = _DivisionCode
            End Get
            Set(ByVal value As String)
                _DivisionCode = value
            End Set
        End Property
        Public Property ProductCode() As String
            Get
                ProductCode = _ProductCode
            End Get
            Set(ByVal value As String)
                _ProductCode = value
            End Set
        End Property
        Public Property Description() As String
            Get
                Description = _Description
            End Get
            Set(ByVal value As String)
                _Description = value
            End Set
        End Property
        Public Property IsOpen() As Nullable(Of Integer)
            Get
                IsOpen = _IsOpen
            End Get
            Set(ByVal value As Nullable(Of Integer))
                _IsOpen = value
            End Set
        End Property

#End Region

#Region " Methods "

        Public Shadows Function ToString(ByVal WithDescription As Boolean) As String

            If WithDescription Then

                ToString = CStr(AdvantageFramework.StringUtilities.PadWithCharacter(Me.Number, 6, "0", True, True) & " - " & Me.Description).Trim

            Else

                ToString = CStr(AdvantageFramework.StringUtilities.PadWithCharacter(Me.Number, 6, "0", True, True)).Trim

            End If

        End Function
        Public Sub New()


        End Sub
        Public Sub New(ByVal Job As Object)

            MyBase.SetValues(Job)

        End Sub

#End Region

    End Class

End Namespace

File: AdvantageFramework.Core/Database/Core/Job.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdvantageFramework.Core.Database.Core
{
    [Serializable()]
    public partial class Job //: AdvantageFramework.BaseClasses.BaseCore
    {

        #region  Constants 



        #endregion

        #region  Enum 

        public enum Properties
        {
            Number,
            OfficeCode,
            ClientCode,
            DivisionCode,
            ProductCode,
            Description,
            IsActive
        }

        #endregion

        #region  Variables 

        private int _Number = default;
        private string _OfficeCode = null;
        private string _ClientCode = null;
        private string _DivisionCode = null;
        private string _ProductCode = null;
        private string _Description = null;
        private int? _IsOpen = default;

        #endregion

        #region  Properties 

        public int Number
        {
            get
            {
                int NumberRet = default;
                NumberRet = _Number;
                return NumberRet;
            }

            set
            {
                _Number = value;
            }
        }

        public string OfficeCode
        {
            get
            {
                string OfficeCodeRet = default;
                OfficeCodeRet = _OfficeCode;
                return OfficeCodeRet;
            }

            set
            {
                _OfficeCode = value;
            }
        }

        public string ClientCode
        {
            get
            {
                string ClientCodeRet = default;
                ClientCodeRet = _ClientCode;
                return ClientCodeRet;
            }

            set
            {
                _ClientCode = value;
            }
        }

        public string DivisionCode
        {
            get
            {
                string DivisionCodeRet = default;
                DivisionCodeRet = _DivisionCode;
                return DivisionCodeRet;
            }

            set
            {
                _DivisionCode = value;
            }
        }

        public string ProductCode
        {
            get
            {
                string ProductCodeRet = default;
                ProductCodeRet = _ProductCode;
                return ProductCodeRet;
            }

            set
            {
                _ProductCode = value;
            }
        }

        public string Description
        {
            get
            {
                string DescriptionRet = default;
                DescriptionRet = _Description;
                return DescriptionRet;
            }

            set
            {
                _Description = value;
            }
        }

        public int? IsOpen
        {
            get
            {
                int? IsOpenRet = default;
                IsOpenRet = _IsOpen;
                return IsOpenRet;
            }

            set
            {
                _IsOpen = value;
            }
        }

        #endregion

        #region  Methods 

        public string ToString(bool WithDescription)
        {
            string ToStringRet = default;
            if (WithDescription)
            {
                ToStringRet = (AdvantageFramework.Core.StringUtilities.Methods.PadWithCharacter(Number.ToString(), 6, "0", true, true) + " - " + Description).Trim();
            }
            else
            {
                ToStringRet = AdvantageFramework.Core.StringUtilities.Methods.PadWithCharacter(Number.ToString(), 6, "0", true, true).Trim();
            }

            return ToStringRet;
        }

        public Job()
        {
        }

        public Job(object Job)
        {
            //base.SetValues(Job);
        }

        #endregion

    }
}
